core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala

```diff
@@ -1001,32 +1041,224 @@ object TLVPoint extends Factory[TLVPoint] {
     val value = UInt64(bytes.drop(1 + outcome.byteSize).take(8))
     val extraPrecision = UInt16(bytes.drop(9 + outcome.byteSize).take(2)).toInt
 
-    TLVPoint(outcome = outcome.toLong,
-             value = Satoshis(value.toLong),
-             extraPrecision = extraPrecision,
-             isEndpoint = isEndpoint)
+    OldTLVPoint(outcome = outcome.toLong,
+                value = Satoshis(value.toLong),
+                extraPrecision = extraPrecision,
+                isEndpoint = isEndpoint)
+  }
+}
+
+sealed trait PayoutCurvePieceTLV extends DLCSetupPieceTLV
+
+object PayoutCurvePieceTLV extends TLVParentFactory[PayoutCurvePieceTLV] {
+
+  override val allFactories: Vector[TLVFactory[PayoutCurvePieceTLV]] =
+    Vector(PolynomialPayoutCurvePieceTLV, HyperbolaPayoutCurvePieceTLV)
+
+  override val typeName: String = "PayoutCurvePieceTLV"
+}
+
+case class PolynomialPayoutCurvePieceTLV(midpoints: Vector[TLVPoint])
+    extends PayoutCurvePieceTLV {
+  override val tpe: BigSizeUInt = PolynomialPayoutCurvePieceTLV.tpe
+
+  override val value: ByteVector = {
+    u16PrefixedList(midpoints)
   }
 }
 
+object PolynomialPayoutCurvePieceTLV
+    extends TLVFactory[PolynomialPayoutCurvePieceTLV] {
+  override val tpe: BigSizeUInt = BigSizeUInt(42792)
+
+  override def fromTLVValue(
+      value: ByteVector): PolynomialPayoutCurvePieceTLV = {
+    val iter = ValueIterator(value)
+
+    val points = iter.takeU16PrefixedList(() => iter.take(TLVPoint))
+
+    PolynomialPayoutCurvePieceTLV(points)
+  }
+
+  override val typeName: String = "PolynomialPayoutCurvePieceTLV"
+}
+
+case class Signed16PTLVNumber(
+    sign: Boolean,
+    withoutPrecision: Long,
+    extraPrecision: Int)
+    extends NetworkElement {
+
+  lazy val toBigDecimal: BigDecimal = {
+    val absVal = withoutPrecision + (BigDecimal(extraPrecision) / (1 << 16))
+
+    if (sign) absVal else -absVal
+  }
+
+  lazy val signByte: Byte = if (sign) {
+    1.toByte
+  } else {
+    0.toByte
+  }
+
+  override def bytes: ByteVector = {
+    ByteVector(signByte) ++
+      BigSizeUInt(withoutPrecision).bytes ++
+      UInt16(extraPrecision).bytes
+  }
+}
+
+object Signed16PTLVNumber extends Factory[Signed16PTLVNumber] {
+
+  override def fromBytes(bytes: ByteVector): Signed16PTLVNumber = {
+    val sign = bytes.head match {
+      case 0 => false
+      case 1 => true
```

Comment:
should use `ValueIterator` here you can can just do `getBool`, if not should do `case FALSE_BYTE` and `case TRUE_BYTE`